pkg/operator/controller/nodeobservabilityrun/error.go

```diff
@@ -0,0 +1,23 @@
+package nodeobservabilityruncontroller
+
+import "fmt"
+
+type NodeObservabilityRunError struct {
+	HttpCode int
+	Msg      string
+}
+
+func (e NodeObservabilityRunError) Error() string {
+	return fmt.Sprintf("Code: %d, Error: %s", e.HttpCode, e.Msg)
+}
+
+func IsNodeObservabilityRunErrorRetriable(err error) bool {
+	e, ok := err.(NodeObservabilityRunError)
+	if !ok {
+		return false
+	}
+	if e.HttpCode == 500 {
```

Comment:
use constant from net/http

Reply:
when the agent returns 500, in most cases, with the current implementation, it is "stuck" until an admin cleans up the error file.
I don't know if we want to keep retrying in this case.
Do we want to look at httpCode 409? It's returned when the agent is busy by another profiling request, and will be able to run this request a bit later.